zmon_agent/discovery/kubernetes/cluster.py

```diff
@@ -289,13 +286,11 @@ def get_cluster_pods_and_containers(
     pods = get_all(kube_client, kube_client.get_pods, namespace, span=current_span)
 
     for pod in pods:
-        if not pod.ready:
-            continue
 
         obj = pod.obj
 
         containers = obj['spec'].get('containers', [])
-        container_statuses = {c['name']: c for c in obj['status']['containerStatuses']}
+        container_statuses = {c['name']: c for c in obj['status'].get('containerStatuses', {})}
         conditions = {c['type']: c['status'] for c in obj['status']['conditions']}
```

Comment:
I'm not sure if `conditions` will be present in a completely new pod. Might be safer to use `get('conditions', [])`.

Reply:
My tests showed they exist. But I think it makes sense to be safe here.